services/catalog/src/simcore_service_catalog/db/repositories/services.py

```diff
@@ -141,7 +141,10 @@ async def list_service_releases(
             async for row in await conn.stream(query):
                 releases.append(ServiceMetaDataAtDB(**row))
 
-        return releases
+        # Now sort naturally from latest first: (This is lame, the sorting should be done in the db)
+        return sorted(
+            releases, key=lambda x: [int(v) for v in x.version.split(".")], reverse=True
```

Comment:
maybe have a look at line 371 (this should be I think the real solution, integrating semver in postgres such that it understands versions).
And for the sorting you could use ```sorted(releases, key=lambda x: packaging.version.parse(x.version)) ``` (not sure about reverse in this case) but it should be more robust to whatever version

Reply:
thanks, much cleaner and reverse works